firestore/integration_test_internal/src/firestore_integration_test.cc

```diff
@@ -280,18 +272,12 @@ FirestoreIntegrationTest::QuerySnapshotToMap(
   return result;
 }
 
-/* static */
-void FirestoreIntegrationTest::Await(const Future<void>& future) {
-  while (future.status() == FutureStatus::kFutureStatusPending) {
```

Comment:
Previously, the overload of `Await` taking a `Future<void>` didn't have a timeout, unlike its `Future<T>` counterpart. Adding a timeout revealed that some of our transaction tests that do retries can take a varying amount of time, depending on the exact values of exponential backoff during that run, that just happens to be within a range of 11-18 seconds. The default timeout is 15 seconds, so that resulted in flaky tests. Adding a way to override the default timeout fixes the issue.